lib/actions/tunnel.ts

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2016-2017 Balena
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as Bluebird from 'bluebird';
+import * as _ from 'lodash';
+import { CommandDefinition } from 'capitano';
+import { stripIndent } from 'common-tags';
+import { isArray } from 'util';
+import { createServer } from 'net';
+import { tunnelConnectionToDevice } from '../utils/tunnel';
+
+interface Args {
+	uuid: string;
+}
+
+interface Options {
+	port: string | string[];
+}
+
+class InvalidPortMappingError extends Error {
+	constructor(mapping: string) {
+		super(`'${mapping}' is not a valid port mapping.`);
+	}
+}
+
+export const tunnel: CommandDefinition<Args, Options> = {
+	signature: 'tunnel [uuid]',
+	description: 'Tunnel local ports to your balenaOS device',
+	help: stripIndent`
+		Use this command to open local ports which tunnel to listening ports on your balenaOS device.
+
+        For example, you could open port 8080 on your local machine to connect to your managed balenaOS
+        device running a web server listening on port 3000.
+
+		You can tunnel multiple ports at any given time.
```

Comment:
Indentation went off a bit here and below.

Reply:
Oops! I'll sort that, good spot.